src/flymake-collection.el

```diff
@@ -81,6 +81,14 @@ like output from OUTPUT into a list and then returns it."
         (forward-line)))
     objects))
 
+(defun flymake-collection-npm-bin (command)
```

Comment:
I really don't like this search up until you see this directory and then look for binaries within it pattern that node and venv style projects have popularised :cry:. Could we use npx or some other tool to execute the linter so we aren't tied to how individual projects decide to lay out dependencies.

Reply:
That means you need to check `npx eslint -v`, then `pnpx eslint -v`, then `yarn eslint -v `. ..., why not just to check binary exist or not directly ? After all, all the tools save the binary in the same directory project-root/node_modules/.bin. I think this is the best solution for now.

Reply:
I agree. checking for the command existing would be more of a pain with this :cry:. But also I don't think it should be this tools responsibility to check extra possible install paths. The user should use something like direnv or [add-node-modules-path](https://github.com/codesuki/add-node-modules-path) to add it for all projects and tools rather than expect each emacs extension to independently do the same check :thinking:. Would that extension solve this same problem for you?

> why not just to check binary exist or not directly ?

To be clear this isn't checking just for the binary existing, it's adding an extra search on each linter invocation in each buffer that can go from the cwd to root just for a node_modules folder just so we can then further search within a sub-directory of that folder for a binary. On a modern Linux machine that's quick and easy but over tramp it can be painfully slow and blocking. This isn't a problem this package needs to solve, it requires something else atop it. direnv and [emacs-direnv](https://github.com/wbolster/emacs-direnv) would be my recommendation. You could also use dir-locals appending to exec-path or any other solution that doesn't require hacking extensions to be aware of how tools are installed.